fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexOptions.java

```diff
@@ -97,6 +97,14 @@ public class LuceneIndexOptions {
      * merge with another adjacent partition.
      */
     public static final String INDEX_PARTITION_LOW_WATERMARK = "partitionLowWatermark";
+    /**
+     * Whether a separate B-tree index gives a direct mapping of live documents to segment by record primary key, that
+     * utilizes an optimized stored fields to manage merges.
+     * Boolean string ({@code true} or {@code false}).
+     * This is incompatible with {@link #PRIMARY_KEY_SEGMENT_INDEX_ENABLED} and {@link #OPTIMIZED_STORED_FIELDS_FORMAT_ENABLED};
+     * if either of those are enabled, this cannot.
```

Comment:
What I found confusing here is that it seems to imply that using V2 means you have to forego optimized stored fields. But what is really going on is that with V2 you have no _choice_ about the matter. Maybe that just needs to be captured in this comment.